packages/backend/server/src/core/auth/resolver.ts

```diff
@@ -48,7 +49,8 @@ export class AuthResolver {
   constructor(
     private readonly config: Config,
     private readonly auth: AuthService,
-    private readonly token: TokenService
+    private readonly token: TokenService,
+    private readonly captcha?: CaptchaService
```

Comment:
I think the core module should not be aware of plugin modules
let me find a better solution for this

Reply:
how do you think about this usage:

``` ts
// register at global
CHALLENGES.register(Captcha);

// guard request in resolver
@UseChallengeGuard()
async somethingWithChallenge() {
}
```

Reply:
or, maybe it's ok to make captcha a standard service and let cloudflare captcha override it.
the captcha verification could receive the whole headers as parameter instead of specified parameters.

Reply:
it's cool but it doesn't has to be registered manually. the injection could be `@Optional()`.
and also I don't like the way which by extending the args of APIs need to be protected

Reply:
because the interface has been removed, the problem no longer exists